ssl/quic/quic_channel.c

```diff
@@ -295,23 +295,44 @@ static int ch_init(QUIC_CHANNEL *ch)
 
     ossl_quic_tx_packetiser_set_ack_tx_cb(ch->txp, ch_on_txp_ack_tx, ch);
 
-    qrx_args.libctx             = ch->port->engine->libctx;
-    qrx_args.demux              = ch->port->demux;
-    qrx_args.short_conn_id_len  = rx_short_dcid_len;
-    qrx_args.max_deferred       = 32;
+    /*
+     * qrx does not exist yet, then we must be dealing with client channel
+     * (QUIC connection initiator).
+     * If qrx exists already, then we are dealing with server channel which
+     * qrx gets created by port_default_packet_handler() before
+     * port_default_packet_handler() accepts connection and creates channel
+     * for it.
+     * The exception here is tserver which always creates channel,
+     * before the first packet is ever seen.
+     */
+    if ((ch->qrx == NULL) && (ch->is_tserver_ch == 0)) {
```

Comment:
IMO these extra parentheses do not improve the readability but I won't insist on removing them

Reply:
OK, will remove them.